autosklearn/ensembles/multiobjective_dummy_ensemble.py

```diff
@@ -0,0 +1,458 @@
+    ----------
+    task_type: int
+        An identifier indicating which task is being performed.
+
+    metrics: Sequence[Scorer] | Scorer
+        The metrics used to evaluate the models.
+
+    random_state: Optional[int | RandomState] = None
+        Not used.
+
+    backend : Backend
+        Gives access to the backend of Auto-sklearn. Not used.
+
+    model_index : int
+        Index of the model that constitutes the ensemble. This index will
+        be used to select the correct predictions that will be passed during
+        ``fit`` and ``predict``.
+    """
+
+    def __init__(
+        self,
+        task_type: int,
+        metrics: Sequence[Scorer] | Scorer,
+        random_state: int | np.random.RandomState | None,
+        backend: Backend,
+        model_index: int,
+    ):
+        # Add some default values -- at least 1 model in ensemble is assumed
+        self.indices_ = [model_index]
+        self.weights_ = [1.0]
+        self.task_type = task_type
+        self.metrics = metrics if isinstance(metrics, Sequence) else [metrics]
+
+    def fit(
+        self,
+        base_models_predictions: np.ndarray | List[np.ndarray],
+        X_data: SUPPORTED_FEAT_TYPES,
+        true_targets: np.ndarray,
+        model_identifiers: List[Tuple[int, int, float]],
+        runs: Sequence[Run],
+    ) -> SingleModelEnsemble:
+        """Dummy implementation of the ``fit`` method.
+
+        Actualy work of passing the model index is done in the constructor. This
+        method only stores the identifier of the selected model and computes it's
+        validation loss.
+
+         Parameters
+         ----------
+         base_models_predictions: np.ndarray
+             shape = (n_base_models, n_data_points, n_targets)
+             n_targets is the number of classes in case of classification,
+             n_targets is 0 or 1 in case of regression
+
+             Can be a list of 2d numpy arrays as well to prevent copying all
+             predictions into a single, large numpy array.
+
+         X_data : list-like or sparse data
+
+         true_targets : array of shape [n_targets]
+
+         model_identifiers : identifier for each base model.
+             Can be used for practical text output of the ensemble.
+
+         runs: Sequence[Run]
+             Additional information for each run executed by SMAC that was
+             considered by the ensemble builder. Not used.
+
+         Returns
+         -------
+         self
+        """
+        self.identifiers_ = [model_identifiers[self.indices_[0]]]
+        loss = calculate_losses(
+            solution=true_targets,
+            prediction=base_models_predictions[self.indices_[0]],
+            task_type=self.task_type,
+            metrics=self.metrics,
+            X_data=X_data,
+        )
+        self.best_model_score_ = loss[self.metrics[0].name]
+        return self
+
+    def predict(self, predictions: Union[np.ndarray, List[np.ndarray]]) -> np.ndarray:
```

Comment:
```suggestion
    def predict(self, predictions: np.ndarray | list[np.ndarray]) -> np.ndarray:
```

Reply:
I applied `pyupgrade`, this should be fixed now.